src/System.Security.Cryptography.OpenSsl/src/System/Security/Cryptography/ECDsaOpenSsl.ImportExport.cs

```diff
@@ -0,0 +1,270 @@
+            return ecparams;
+        }
+
+        private static ECParameters ExportNamedCurveParameters(SafeEcKeyHandle key, bool includePrivate)
+        {
+            // It's entirely possible that this line will cause the key to be generated in the first place.
+            CheckInvalidKey(key);
+
+            ECParameters parameters = new ECParameters();
+
+            Interop.Crypto.GetECKeyParameters(key, includePrivate, out parameters.Q.X, out parameters.Q.Y, out parameters.D);
+
+            bool hasPrivateKey = (parameters.D != null);
+            if (hasPrivateKey != includePrivate) // || !HasConsistentPrivateKey(rsaParameters))
+            {
+                throw new CryptographicException(SR.Cryptography_CSP_NoPrivateKey);
+            }
+
+            return parameters;
+        }
+
+        private static ECParameters ExportExplicitCurveParameters(SafeEcKeyHandle key, bool includePrivate)
+        {
+            // It's entirely possible that this line will cause the key to be generated in the first place.
+            CheckInvalidKey(key);
+
+            ECCurveType curveType = Interop.Crypto.EcKeyGetCurveType(key);
+            if (curveType == ECCurveType.Unspecified)
+            {
+                throw new PlatformNotSupportedException();
+            }
+
+            ECParameters parameters = new ECParameters();
+            if (curveType == ECCurveType.Characteristic2)
+            {
+                var curve = new ECCharacteristic2Curve();
+                parameters.Curve = curve;
+                byte[] p;
+
+                Interop.Crypto.GetECCurveParameters(key, includePrivate, out curve.CurveType, out parameters.Q.X, out parameters.Q.Y, out parameters.D,
+                    out p, out curve.A, out curve.B, out curve.G.X, out curve.G.Y, out curve.N, out curve.H, out curve.S);
+                curve.M = p;//[0];
+            }
+            else
+            {
+                var curve = new ECPrimeCurve();
+                parameters.Curve = curve;
+
+                Interop.Crypto.GetECCurveParameters(key, includePrivate, out curve.CurveType, out parameters.Q.X, out parameters.Q.Y, out parameters.D,
+                    out curve.P, out curve.A, out curve.B, out curve.G.X, out curve.G.Y, out curve.N, out curve.H, out curve.S);
+            }
+
+
+            bool hasPrivateKey = (parameters.D != null);
+            if (hasPrivateKey != includePrivate) // || !HasConsistentPrivateKey(rsaParameters))
+            {
+                throw new CryptographicException(SR.Cryptography_CSP_NoPrivateKey);
+            }
+
+            return parameters;
+        }
+
+        private static SafeEcKeyHandle ImportNamedCurveParameters(ECParameters parameters, ECNamedCurve curve, bool includePrivate)
+        {
+            if (curve.Oid == null)
+                throw Interop.Crypto.CreateOpenSslCryptographicException(); // todo arg exception?
```

Comment:
I'd expect this one to be an assert, really; and that the ValidateNamedCurve method checked that part.